File: GT-Models/src/main/java/com/DavideDalSanto/GTModels/Entities/Workout.java
package com.DavideDalSanto.GTModels.Entities;

import lombok.*;

import javax.persistence.*;
import java.util.List;

@AllArgsConstructor
@NoArgsConstructor
@Getter
@Setter
@Entity
@Table(name = "workouts")
public class Workout {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String name;

    @ManyToMany
    @JoinTable(name = "workouts_user_exercises",
            joinColumns = @JoinColumn(name = "workout_id"),
            inverseJoinColumns = @JoinColumn(name = "user_exercise_id"))
    private List<UserExercise> userExercises = new java.util.ArrayList<>();


}

File: FE/GymTonic-FE/src/app/Models/workout.ts
import { Exercise } from "./exercise";
import { UserExercise } from "./user-exercise";

export class Workout {
  id!:number;
  name:string;
  userExercises:UserExercise[] = [];

  constructor(name:string, userExs:UserExercise[]){
    this.name = name;
    this.userExercises = userExs;

  }

}
